### PR TITLE
test(dual-query-planner): added description to semantic diff failures

### PR DESCRIPTION
**Pull request by duckki**
- With diff description, it's slightly easier to identify where the difference is.
- This is a facility for testing.

---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [x] Changes are compatible[^1]
- [ ] Documentation[^2] completed
- [x] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [ ] Unit Tests
    - [ ] Integration Tests
    - [x] Manual Tests

[ROUTER-470]: https://apollographql.atlassian.net/browse/ROUTER-470?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apollo-router/src/query_planner/dual_query_planner.rs [hunk 1]
```diff
@@ -318,25 +403,65 @@ fn vec_matches_as_set<T>(this: &[T], other: &[T], item_matches: impl Fn(&T, &T)
...
+                .map_err(|err| err.add_description("under Sequence node"))?;
         }
         (PlanNode::Parallel { nodes: this }, PlanNode::Parallel { nodes: other }) => {
-            vec_matches_as_set(this, other, plan_node_matches)
+            vec_matches_result_as_set(this, other, |a, b| plan_node_matches(a, b).is_ok())
+                .map_err(|err| err.add_description("under Parallel node"))?;
+        }
+        (PlanNode::Fetch(this), PlanNode::Fetch(other)) => {
+            fetch_node_matches(this, other)
+                .map_err(|err| err.add_description("under Fetch node"))?;
```
Comment (goto-bus-stop) [inline apollo-router/src/query_planner/dual_query_planner.rs:450]: Might be nice to add the `this.operation_name` to the message

**Review by goto-bus-stop — APPROVED** [review]
I only skimmed it, at that level it looks good to me 😄

File: apollo-router/src/query_planner/dual_query_planner.rs (hunk 1 above)
Comment (duckki) [reply apollo-router/src/query_planner/dual_query_planner.rs:450]: good idea!

Comment (goto-bus-stop) [reply apollo-router/src/query_planner/dual_query_planner.rs:450]: Or would it make it harder to categorise differences? It's nice to have for an individual plan but maybe not for mass-comparisons with harness.

Comment (duckki) [reply apollo-router/src/query_planner/dual_query_planner.rs:450]: I think I can filter out operation names in the classification script.

**Pull request merged**